nh_clinical/activity_extension.py

```diff
@@ -620,10 +620,8 @@ def trigger_policy(self, cr, uid, activity_id, location_id=None,
                 'creator_id': activity_id
             }, data, context=context)
             if trigger_activity['type'] == 'recurring':
-                frequency = activity_pool.browse(
-                    cr, SUPERUSER_ID, ta_activity_id,
-                    context=context).data_ref.frequency
-                date_schedule = (dt.now()+td(minutes=frequency)).strftime(DTF)
+                date_schedule = self.get_recurring_activity_date_scheduled(
+                    cr, uid, ta_activity_id, context)
             else:
                 date_schedule = dt.now()+td(minutes=60)
```

Comment:
What's this 60 minutes value do?

Reply:
Is it worth dealing with the complexity of the trigger policy method?

Reply:
Discussed this and unfortunately I have nothing valuable to add to [existing tech debt](https://jupiter.bjss.com/browse/EOBS-703).

Reply:
@moderatemisbehaviour @Gimpneek - Is there tech debt to deal with the `else` statement Colin mentioned above? As far as I know, that line is never being hit. EOBS-703 deals with the complexity but I am unsure if this would be covered by that at all.

Reply:
I agree that this entire method needs refactoring (I raised the tech debt for it) but I don't see what's wrong with the `else` statement until that refactoring happens. It will be hit if `nh.clinical.patient.admission.trigger_policy` is ever called which I assume it will be. [You can see the policy doesn't have a type of `recurring`](https://github.com/NeovaHealth/openeobs/blob/master/nh_eobs/policy.py#L46) so it should hit that `else`.